INTRODUCTION.md

```diff
@@ -1540,6 +1541,96 @@ the original fatal error code and reason.
 To read more about static group membership, see [KIP-345](https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances).
 
 
+### Next generation of the consumer group protocol: [KIP 848](https://cwiki.apache.org/confluence/display/KAFKA/KIP-848%3A+The+Next+Generation+of+the+Consumer+Rebalance+Protocol)
+
+Starting from librdkafka 2.4.0 the next generation consumer group rebalance protocol
+is in **Early Access**. It means it's still  _not production-ready_ and
+_not supported_, given it's still under validation and lacking some needed features.
+
+With this protocol the role of the Group Leader (a member) is removed and
+the assignment is calculated by the Group Coordinator (a broker) and sent
+to each member through heartbeats.
+
+To test it, a Kafka cluster must be set up, in KRaft mode, and the new group
+protocol enabled with the `group.coordinator.rebalance.protocols` property.
+Broker version must be Apache Kafka 3.7.0 or newer.
+
+Client side, it can be enabled by setting the new property `group.protocol=consumer`.
+A second property named `group.remote.assignor` is added to choose desired
+remote assignor.
+
+**Available features**
+
+- Subscription to one or more topics
+- Rebalance callbacks (see contract changes)
+- Static group membership
+- Configure remote assignor
+- Max poll interval is enforced
+- Offline upgrade from an empty consumer group with committed offsets
+
+**Missing features**
+
+- Regular expression support when subscribing
+- AdminClient changes as described in the KIP
+- Client side assignors as described in the KIP
+- Online upgrade of a non-empty consumer group
+
+**Contract changes**
+
+Along with the new feature there are some needed contract changes,
+so the protocol will be enabled by default only with a librdkafka major release.
+
+ - These client properties are deprecated: `partition.assignment.strategy`,
+ replaced by `group.remote.assignor`;
+`session.timeout.ms` and `heartbeat.interval.ms`, replaced by broker configurations
+`group.consumer.session.timeout.ms` and `group.consumer.heartbeat.interval.ms`.
+
+ - Protocol rebalance is fully incremental, so the only allowed functions to
+   use in a rebalance callback will be `rd_kafka_incremental_assign` and
+   `rd_kafka_incremental_unassign`. Currently you can still use existing code
+   and the expected function to call is determined based on the chosen
+   `partition.assignment.strategy` but this will be removed already in next
+   release.
```

Comment:
```suggestion
 - Protocol rebalance is fully incremental, so the only allowed functions to
   use in a rebalance callback will be `rd_kafka_incremental_assign` and
   `rd_kafka_incremental_unassign`. Currently you can still use existing code
   and the expected function to call is determined based on the chosen
   `partition.assignment.strategy` but this will be removed in the next
   release.
```